sw/device/lib/dif/dif_spi_device.c

```diff
@@ -0,0 +1,686 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_spi_device.h"
+
+#include "spi_device_regs.h"  // Generated.
+#include "sw/device/lib/base/memory.h"
+
+const uint16_t kDifSpiDeviceSramLen = SPI_DEVICE_BUFFER_SIZE_BYTES;
+
+/**
+ * Computes the required value of the control register from a given
+ * configuration.
+ */
+static dif_spi_device_result_t build_control_word(
+    const dif_spi_device_config_t *config, uint32_t *device_config) {
+  *device_config = 0;
+  switch (config->clock_polarity) {
+    case kDifSpiDeviceEdgePositive:
+      *device_config |= 0x0 << SPI_DEVICE_CFG_CPOL;
+      break;
+    case kDifSpiDeviceEdgeNegative:
+      *device_config |= 0x1 << SPI_DEVICE_CFG_CPOL;
+      break;
+    default:
+      return kDifSpiDeviceResultBadArg;
+  }
+  switch (config->data_phase) {
+    case kDifSpiDeviceEdgePositive:
+      *device_config |= 0x1 << SPI_DEVICE_CFG_CPHA;
+      break;
+    case kDifSpiDeviceEdgeNegative:
+      *device_config |= 0x0 << SPI_DEVICE_CFG_CPHA;
+      break;
+    default:
+      return kDifSpiDeviceResultBadArg;
+  }
+  switch (config->tx_order) {
+    case kDifSpiDeviceBitOrderMsbToLsb:
+      *device_config |= 0x0 << SPI_DEVICE_CFG_TX_ORDER;
+      break;
+    case kDifSpiDeviceBitOrderLsbToMsb:
+      *device_config |= 0x1 << SPI_DEVICE_CFG_TX_ORDER;
+      break;
+    default:
+      return kDifSpiDeviceResultBadArg;
+  }
+  switch (config->rx_order) {
+    case kDifSpiDeviceBitOrderMsbToLsb:
+      *device_config |= 0x0 << SPI_DEVICE_CFG_RX_ORDER;
```

Comment:
I mean, sure. The no-op is intended to be like... to communicate intent? Do you think it would be better to just do `if (config->tx_order == kPositive) { ... }` rather than all these silly switches?

Reply:
>Why not just omit them?

I think in this case that could be the best option? Especially if you stick `reset` routine before the configuration, you would ensure that all of the registers are reset anyway.

Reply:
Done.